src/lang-promql/complete/hybrid.ts

```diff
@@ -94,46 +96,20 @@ const parsedSnippets = snippets.map((s) => ({
   score: 0,
 }));
 
-function arrayToCompletionResult(
-  data: AutoCompleteNode[],
-  from: number,
-  to: number,
-  context: AutocompleteContext,
-  state: EditorState,
-  includeSnippet = false
-): CompletionResult {
-  const text = state.sliceDoc(from, to);
-  const options: Completion[] = [];
-  for (const completionList of data) {
-    for (const label of completionList.labels) {
-      let score: number | null;
-      if ((score = context.filter(label, text, true))) {
-        options.push({ label: label, apply: '', type: completionList.type, score: score });
-      }
-    }
-  }
-  if (includeSnippet) {
-    for (const s of parsedSnippets) {
-      let score: number | null;
-      if ((score = context.filter(s.label, text, false))) {
-        s.score = score;
-        options.push(s);
-      }
-    }
-  }
-  return {
-    from: from,
-    to: to,
-    options: options,
-  } as CompletionResult;
+function escape(text: string): string {
+  return text.replace(/</g, '&lt;').replace(/>/g, '&gt;');
```

Comment:
You'll need to replace at least `&` as well, and maybe better also a couple of other chars. This is the `escapeHTML()` function from Prometheus:

https://github.com/prometheus/prometheus/blob/f97d2ddb6e6fce1343d2b026479d19592b934a36/web/ui/react-app/src/utils/index.ts#L16-L29

Reply:
Ah ok. Funny things btw, looks like the map is worst choice for this use case.
https://stackoverflow.com/a/28458409

No Idea how the Switch case ban be better than a map.
I supposed you lost time when creating the map

Reply:
Interesting... though the differences are so minor that it probably doesn't matter. The worst case (several regex replaces) is "2% slower" in my browser.

Reply:
yup it doesn't really matter, I just found it funny ;)

Reply:
fixed